Nick_Maven/ErrorExeptions/ProjectPackage/Student.java

```diff
@@ -0,0 +1,132 @@
+package Nick_Maven.ErrorExeptions.ProjectPackage;
+
+public class Student extends Group {
+    public Subject[] subject;
+    private String studentName;
+
+    public Student(String facultyName, String groupName, String studentName, Subject[] subject) {
+        super(facultyName, groupName);
+        try {
+            if (studentName.length() <= 1) {
+                throw new StudentNameException("Some of Students don't have name. I skip him");
+            }
+            this.studentName = studentName;
+        } catch (StudentNameException ex) {
+            this.setStudentName("UNNAMED");
+        }
+        try {
+            if (subject.length < 1) {
+                throw new SubjectCountException("Every student must have at least ONE SUBJECT! I add 'UNNAMED' " +
+                        "with grade = 0");
+            }
+            this.subject = subject;
+        } catch (SubjectCountException ex) {
+            this.subject = new Subject[1];
+            this.subject[0] = new Subject("UNNAMED", 0);
+        }
+        try {
+            if (this.getGroupName().length() < 1) throw new GroupNameException("Every student must have group name! I skip him");
+        } catch (GroupNameException ex) {
+            this.setGroupName("UNNAMED");
+        }
+        try {
+            if (this.getFacultyName().length() < 1) throw new FacultyNameException("Every student must have the faculty! I skip him");
+        } catch (FacultyNameException ex4) {
+            this.setFacultyName("UNNAMED");
+        }
+    }
+
+    class StudentNameException extends Exception {
+        public StudentNameException(String message) {
+            super(message);
+            System.out.println(message);
+        }
+    }
+
+    class SubjectCountException extends Exception {
+        public SubjectCountException(String message) {
+            super(message);
+            System.out.println(message);
+        }
+    }
+
+    class GroupNameException extends Exception {
+        public GroupNameException(String message) {
+            super(message);
+            System.out.println(message);
+        }
+    }
+
+    class FacultyNameException extends Exception {
```

Comment:
Create a separate package for exceptions and put all exceptions to this package as divided classes